### ISSUE TITLE
Python program in startup folder does execute on startup

### ISSUE THREAD

**Issue by mrinmoyprasad** [jupyter/notebook#2409]
I have a file startup.ipy placed in /home/jovyan/.ipython/profile_default/startup/ on my docker image (scipy-notebook). startup.ipy contains the following command `% run /home/jovyan/work/sql-wrapper.ipy`

On opening a Python 3 notebook, I do not see the sql-wrapper.ipy being executed. However, running them manually inside a cell or exueting  `% run /home/jovyan/work/sql-wrapper.ipy` works.

I using JupyterHub (0.7.2) and scipy-notebook from docker-stacks as my single user server docker image. I have also stripped  the installation of Python 2 environment from the image as its not really required.

**Comment by takluyver** [comment]
Code running when the kernel starts doesn't show any visible output in the notebook, so it might be that it's running, just not visibly. Can you set a variable or something in the script and see if it has taken effect in the notebook?

**Comment by mrinmoyprasad** [comment]
My python program creates an object like `query` which has available methods like run(), create_table() , etc available. This worked fine with Python 2 notebook. Do I need to make some configurational changes in kernel/ipykernel/conda environment? I have not created any notebook profile here. Do I need to create one? Or do I need to put my startup.ipy file in ~/.jupyter/ folder somewhere?

**Comment by takluyver** [comment]
No,  it should all work the same way. If there's an error, though, you won't see it.

Does it work if you run `ipython` in a terminal using Python 3?

**Comment by mrinmoyprasad** [comment]
So, I tried a couple of things. Hopefully, this helps you explain me what I am doing wrong.

`ipython /home/jovyan/work/sql-wrapper.ipy`
Error: ipython not defined

`% run /home/jovyan/work/sql-wrapper.ipy`
Success

Note: `/home/jovyan/.ipython/profile_default/startup/startup.ipy` contains the following line `% run /home/jovyan/work/sql-wrapper.ipy`

**Comment by ihleonard-c3** [comment]
@mrinmoyprasad Did you ever figure out what needed to be changed? I'm having the same issue.

@takluyver If I run `ipython` from the terminal (logged into the container) then the scripts run and my variables are set. But the startup scripts don't run when opening notebooks. Additionally, I've tested locally (not in the docker container) and things do work properly in jupyter notebook. It seems to be an issue with the docker image somehow. Any ideas?

Thanks

**Comment by takluyver** [comment]
In the notebook where it's not working, check `get_ipython().profile_dir.startup_dir` to see if it's pointing to the location you expect.

**Comment by aarondr77** [comment]
I am running into the same problem. I have a ```file 00-start.py```.

If I run ```ipython``` in my terminal, then my file executes on the print statements inside of it get printed to the terminal. 

However, if instead, I launch Jupyter Lab from my terminal by running the command ```jupyter lab``` then my print statements don't occur and the import statements that are in my file don't execute.
